src/core/action/edit/space_domain/ApplyOtsuThresholdEstimationAction.java

```diff
@@ -0,0 +1,132 @@
+        }
+
+        //calculo la media global igual que la acumulativa, pero t sería 255
+        Double globalCumulativeMean = cumulativeMeans[255];
+
+        //calculo varianza entre clases para cada t: (mg * P1(t) - m(t))² / P1(t) * (1 - P1(t))
+        Double[] variances = new Double[256];
+        for (int t = 0; t <= 255; t++){
+            Double tCumulativeProbability = cumulativeProbabilities[t];
+            Double tCumulativeMean = cumulativeMeans[t];
+            variances[t] = Math.pow(((globalCumulativeMean * tCumulativeProbability) - tCumulativeMean), 2)
+                            / (tCumulativeProbability * (1 - tCumulativeProbability));
+        }
+
+        //busco con que t me dio una varianza mas grande. Pueden ser mas de uno
+        List<Integer> tMaxValues = this.searchTThatGenerateMaximumVariance(variances);
+
+        //veo si es el t es unico o son varios valores. Si es unico el valor de umbral es t, sino calculo la media de todos los t
+        int threshold = 0;
+        if(tMaxValues.size() > 1){
+            Integer sum = 0;
+            for (int i = 0; i < tMaxValues.size(); i++){
+                sum += tMaxValues.get(i);
+            }
+            double aux = (double) sum / tMaxValues.size();
+            threshold = (int) aux;
+        }
+        else{
+            threshold = tMaxValues.get(0);
+        }
+
+        //calculo la forma matricial de la imagen
+        int[][] imageMatrix = this.matrixService.toGrayMatrix(customImage.toFXImage());
+
+        //le aplico el umbral calculado
+        int[][] transformedImage = this.applyThreshold(imageMatrix, threshold);
+
+        //creo un objeto con la imagen modificada y el umbral usado, y lo devuelvo
+        OtsuThresholdImage otsuThresholdImage = new OtsuThresholdImage(this.matrixService.toImage(transformedImage, transformedImage, transformedImage),
+                threshold);
+        return otsuThresholdImage;
+    }
+
+
+    private Double cumulativeProbability(Histogram histogram, int limit) {
+        Double value = 0.0;
+        for (int i = 0; i <= limit; i++) {
+            value += histogram.getValues()[i];
+        }
+        return value / histogram.getTotalPixels();
+    }
+
+    private Double cumulativeMean(Histogram histogram, int limit) {
+        Double value = 0.0;
+        for (int i = 0; i <= limit; i++) {
+            value += ((histogram.getValues()[i]) * i);
+        }
+        return value / histogram.getTotalPixels();
+    }
+
+    private List<Integer> searchTThatGenerateMaximumVariance(Double[] variances){
+        List<Integer> tMaxValues = new ArrayList<Integer>();
+        Double maximumVariance = 0.0;
+        for (int t = 0; t < variances.length; t++){
+            if(variances[t] >= maximumVariance){
+                maximumVariance = variances[t];
+            }
+        }
+
+        for (int t = 0; t < variances.length; t++){
+            if(variances[t] == maximumVariance){
+                tMaxValues.add(t);
```

Comment:
este for me hace ruido. estas metiendo el mismo valor muchas veces? no lo termino de entender, para que serviria. si es solo para calcular despues la media, creo que te daria siempre el mismo numero. 
@SynysterLove ayudame un poco con este metodo. no me termina de cerrar

Reply:
No, no está haciendo eso. Está metiendo todos los t que genera que la varianza de el valor máximo calculado anteriormente